projects/plugins/jetpack/modules/related-posts/jetpack-related-posts.php

```diff
@@ -1669,6 +1686,62 @@ protected function filter_non_public_posts( array $related_posts ) {
 		return $filtered;
 	}
 
+	/**
+	 * Generates a context for the related content (second line in related post output).
+	 * Order of importance:
+	 *   - First category (Not 'Uncategorized')
+	 *   - First post tag
+	 *   - Number of comments
+	 *
+	 * @param int $post_id - the post ID.
+	 * @uses get_the_category, get_the_terms, get_comments_number, number_format_i18n, __, _n
+	 * @return string
+	 */
+	protected function generate_related_post_context_block( $post_id ) {
+		$categories = get_the_category( $post_id );
+		if ( is_array( $categories ) ) {
+			foreach ( $categories as $category ) {
+				$cat_link = get_category_link( $category );
+				if ( 'uncategorized' !== $category->slug && '' !== trim( $category->name ) ) {
```

Comment:
Is the intent here to avoid "Uncategorized", or to avoid the default category (which can be named differently now)?

Reply:
To be honest, I was just aiming to be consistent with the existing behaviour used in the non-block rendering.

I do think this is probably the "correct" behaviour though, in that it will only single out posts based on the slug. If the user has renamed their default category (name & slug), those posts will be recognized. I'm not sure I'd argue that we need to keep this behaviour in the new code path, if you think it makes sense to remove the check entirely.

Reply:
PS: The function comment calls this out as the original intent.

> First category (Not 'Uncategorized')